modules/core/02-client/keeper/client.go

```diff
@@ -122,46 +121,41 @@ func (k Keeper) UpdateClient(ctx sdk.Context, clientID string, clientMsg exporte
 
 // UpgradeClient upgrades the client to a new client state if this new client was committed to
 // by the old client at the specified upgrade height
-func (k Keeper) UpgradeClient(ctx sdk.Context, clientID string, upgradedClient exported.ClientState, upgradedConsState exported.ConsensusState,
-	upgradeClientProof, upgradeConsensusStateProof []byte,
+func (k Keeper) UpgradeClient(
+	ctx sdk.Context,
+	clientID string,
+	upgradedClient, upgradedConsState, upgradeClientProof, upgradeConsensusStateProof []byte,
```

Comment:
Isn't this an API breaking change? shouldn't the API remain the same here in the 02-client keeper and continue to pass the `exported` interfaces.

We can then pass `.Value` bytes when calling into the light client module.

Reply:
It is API breaking, but I think that this refactoring will be API breaking anyway and will be released in a major release. I basically did here [a similar change as was done in `CreateClient`](https://github.com/cosmos/ibc-go/blob/f4ee20bf8c9139c7806c6410785ae37073bc7702/modules/core/02-client/keeper/client.go#L20).